eth2/fork_choice/src/optimised_lmd_ghost.rs

```diff
@@ -212,41 +198,55 @@ where
 
     // Finds the best child, splitting children into a binary tree, based on their hashes
     fn choose_best_child(&self, votes: &HashMap<Hash256, u64>) -> Option<Hash256> {
-        let mut bitmask = 0;
-        for bit in (0..=255).rev() {
+        if votes.is_empty() {
+            return None;
+        }
+        let mut bitmask: BitVec = BitVec::new();
+        // loop through bytes then bits
+        for bit in 0..=256 {
             let mut zero_votes = 0;
             let mut one_votes = 0;
-            let mut single_candidate = None;
+            let mut single_candidate = (None, false);
 
+            trace!("FORKCHOICE: Child vote length: {}", votes.len());
             for (candidate, votes) in votes.iter() {
-                let candidate_uint = BigEndian::read_u32(candidate);
-                if candidate_uint >> (bit + 1) != bitmask {
+                let candidate_bit: BitVec = BitVec::from_bytes(&candidate);
+                /*
```

Comment:
Just checking you intended to leave these comments in.

Reply:
Yeah.. they are really handy for debugging (figure I'll need them when the other guys disclose a full set of test vectors). I've added and removed them a few times.. so now i'm leaving in with comments.